src/Mvc/Mvc.Core/src/ApplicationParts/ApplicationPartManager.cs

```diff
@@ -52,18 +52,56 @@ public void PopulateFeature<TFeature>(TFeature feature)
 
         internal void PopulateDefaultParts(string entryAssemblyName)
```

Comment:
@pranavkm it would be good to profile this to understand how this particular piece of code scales (allocations wise) with the number of assemblies. I know it only happens once per app but I'm curious mostly because of the amount of LINQ

Reply:
Ok. We used to have a ton of Linq in the Deps file code too which is primarily why I didn't think it was a big deal: https://github.com/aspnet/AspNetCore/blob/release/3.0-preview5/src/Mvc/Mvc.Core/src/ApplicationParts/ApplicationAssembliesProvider.cs#L48-L79. We could switch it to vanilla for loops if you think this might be something to be concerned about.